src/shared/storage.ts

```diff
@@ -62,10 +64,51 @@ class ExtensionStorage {
 }
 const storage = new ExtensionStorage(chrome.storage.local, chrome.runtime);
 
-export const persistConfig: PersistConfig<RootState> = {
+// Previously we used redux-persist's serialization feature. As `chrome.storage`
+// doesn't require data to be serialized, we can disable that feature, however
+// it requires a migration.
+// Disabling it has a few benefits:
+// - We rely less on redux-persist's implementation details, such as to parse the store
+// - We can more easily read store data in the background script
+// - We can more easily declare default wallet state for use in tests
+const legacyPersistConfig: PersistConfig<RootState> = {
   key: 'root',
   version: 1,
   storage,
   serialize: true,
   whitelist: ['analytics', 'chains', 'keys', 'networks', 'onboarding', 'settings'],
 };
+
+async function migrateToUsingNoSerialization() {
+  const storageVal = (await new Promise(resolve =>
+    chrome.storage.local.get(['persist:root'], resolve)
+  )) as any;
+
+  if (!storageVal) return undefined;
+
+  const store = storageVal['persist:root'];
+
+  if (typeof store === 'string') {
+    void analytics.track('redux_persist_migration_to_no_serialization');
+    return getStoredState(legacyPersistConfig);
+  }
+
+  return storage;
+}
+
+interface UntypedDeserializeOption {
+  deserialize?: boolean;
+}
+export const persistConfig: PersistConfig<RootState> & UntypedDeserializeOption = {
+  key: 'root',
+  version: 1,
+  storage,
+  serialize: false,
+  migrate: createMigrate({
+    0: async () => {
+      return migrateToUsingNoSerialization();
```

Comment:
Sweet, now I can migrate the poorly added onboarding state. 😆

Reply:
haha, yeah go ahead. This is actually kinda cheating, because we already had version `1`, but because the serialisation renders it unreadable, it considers this a needed migration.

Reply:
if you do, add a test to this file 

https://github.com/hirosystems/wallet/blob/546ef414a6532fe5b7e2cc82e2af63955cde8ef0/tests/specs/store-migrations/store-migrations.spec.ts#L11